nice, this looks good. some tests need to be updated. :+1:  on :white_check_mark: 

```
======================================================================
FAIL: test_prod_config (testapps.test_elasticsearch.tests.test_prod_es_indices.ProdIndexManagementTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/testapps/test_elasticsearch/tests/test_prod_es_indices.py", line 27, in test_prod_config
    self.assertEqual(EXPECTED_PROD_INDICES, found_prod_indices)
AssertionError: Lists differ: [{'alias': 'case_search', 'typ... != [{'alias': u'case_search', 'ty...
First differing element 5:
{'alias': 'hqusers', 'type': 'user', 'index': 'test_hqusers_2016-02-16_1402'}
{'alias': u'hqusers', 'type': u'user', 'index': u'test_hqusers_2016-07-19'}
```
```
======================================================================
ERROR: test_query_no_search_all (corehq.apps.userreports.tests.test_choice_provider.UserChoiceProviderTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock.py", line 1224, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_choice_provider.py", line 106, in test_query_no_search_all
    self._test_query(ChoiceQueryContext('', limit=20, page=0))
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_choice_provider.py", line 96, in _test_query
    self.choice_provider.query(query_context),
  File "/mnt/commcare-hq-ro/corehq/apps/userreports/reports/filters/choice_providers.py", line 187, in query
    limit=query_context.limit, offset=query_context.offset)
  File "/mnt/commcare-hq-ro/corehq/apps/users/analytics.py", line 67, in get_search_users_in_domain_es_query
    .search_string_query(search_string, default_search_fields)
  File "/mnt/commcare-hq-ro/corehq/apps/es/fake/es_query_fake.py", line 69, in search_string_query
    lambda doc: any(doc[field] is not None and (search_string in doc[field])
  File "/mnt/commcare-hq-ro/corehq/apps/es/fake/es_query_fake.py", line 43, in _filtered
    return self._clone(result_docs=filter(filter_function, self._result_docs))
  File "/mnt/commcare-hq-ro/corehq/apps/es/fake/es_query_fake.py", line 70, in <lambda>
    for field in default_fields))
  File "/mnt/commcare-hq-ro/corehq/apps/es/fake/es_query_fake.py", line 70, in <genexpr>
    for field in default_fields))
KeyError: 'base_username'
```

--- a/corehq/apps/userreports/tests/test_choice_provider.py
+++ b/corehq/apps/userreports/tests/test_choice_provider.py
@@ -194,6 +194,7 @@ class UserChoiceProviderTest(SimpleTestCase, ChoiceProviderTestMixin):
         user.domain_membership = DomainMembership(domain=domain)
         doc = user._doc
         doc['username.exact'] = doc['username']
+        doc['base_username'] = username
         UserESFake.save_doc(doc)
         return user
 
@@ -205,6 +206,7 @@ class UserChoiceProviderTest(SimpleTestCase, ChoiceProviderTestMixin):
         user.domain_memberships = [DomainMembership(domain=cls.domain)]
         doc = user._doc
         doc['username.exact'] = doc['username']
+        doc['base_username'] = email
         UserESFake.save_doc(doc)
         return user
 

--- a/testapps/test_elasticsearch/tests/test_prod_es_indices.py
+++ b/testapps/test_elasticsearch/tests/test_prod_es_indices.py
@@ -55,7 +55,7 @@ EXPECTED_PROD_INDICES = [
     },
     {
         "alias": "hqusers",
-        "index": "test_hqusers_2016-02-16_1402",
+        "index": "test_hqusers_2016-07-19",
         "type": "user"
     },
     {

